types/three/examples/jsm/nodes/core/NodeBuilder.d.ts

```diff
@@ -0,0 +1,125 @@
+import {
+    BufferGeometry,
+    Material,
+    Object3D,
+    Renderer,
+    Scene,
+    Texture,
+    TextureEncoding,
+    WebGLRenderTarget,
+} from '../../../../src/Three';
+import FogNode from '../fog/FogNode';
+import LightsNode from '../lighting/LightsNode';
+import { NodeShaderStageOption, NodeTypeOption } from './constants';
+import Node from './Node';
+import NodeAttribute from './NodeAttribute';
+import NodeParser from './NodeParser';
+import NodeUniform from './NodeUniform';
+import NodeVar from './NodeVar';
+import NodeVary from './NodeVary';
+
+export type BuildStageOption = 'construct' | 'analyze' | 'generate';
+
+export interface FlowData {
+    code: string;
+}
+
+export default class NodeBuilder {
+    object: Object3D;
+    material: Material;
+    geometry: BufferGeometry;
+    renderer: Renderer;
+    parser: NodeParser;
+
+    nodes: Node[];
+    updateNodes: Node[];
+    hashNodes: { [hash: string]: Node };
+
+    scene: Scene;
+    lightsNode: LightsNode;
+    fogNode: FogNode;
+
+    vertexShader: string;
+    fragmentShader: string;
+    computeShader: string;
+
+    shaderStage: NodeShaderStageOption | null;
+    buildStage: BuildStageOption | null;
+    stack: Node[];
+    get node(): Node;
+
+    addStack(node: Node): void;
+    removeStack(node: Node): void;
+    setHashNode(node: Node, hash: string): void;
+    addNode(node: Node): void;
+    getMethod(method: string): string;
+    getNodeFromHash(hash: string): Node;
+
+    addFlow(shaderStage: NodeShaderStageOption, node: Node): Node;
+
+    setContext(context: any): void;
+    getContext(): any;
```

Comment:
I think `context` is also supposed to be an object

Reply:
Yes. I will change it.